tests/unit/acquisition/test_rule.py

```diff
@@ -817,3 +818,77 @@ def test_asynchronous_rule_add_pending_points() -> None:
     state = AsynchronousRuleState(tf.constant([[1, 1], [2, 2]]))
     state = state.add_pending_points(tf.constant([[3, 3], [4, 4]]))
     npt.assert_array_equal(state.pending_points, [[1, 1], [2, 2], [3, 3], [4, 4]])
+
+
+@pytest.mark.parametrize(
+    "batch_size,ga_population_size,ga_n_generations,filter_threshold",
+    [
+        (-2, 500, 200, 0.1),
+        (0, 500, 200, 0.1),
+        (10, -2, 200, 0.1),
+        (10, 0, 200, 0.1),
+        (10, 500, -2, 0.1),
+        (10, 500, 0, 0.1),
+        (10, 500, 200, -0.1),
+        (10, 500, 200, 1.1),
+    ],
+)
+def test_qhsri_raises_invalid_parameters(
+    batch_size: int, ga_population_size: int, ga_n_generations: int, filter_threshold: float
+) -> None:
+
+    with pytest.raises(ValueError):
+        BatchHypervolumeSharpeRatioIndicator(
+            batch_size, ga_population_size, ga_n_generations, filter_threshold
+        )
+
+
+@pytest.mark.parametrize(
+    "models",
+    [
+        {},
+        {"foo": QuadraticMeanAndRBFKernel()},
+        {"foo": QuadraticMeanAndRBFKernel(), OBJECTIVE: QuadraticMeanAndRBFKernel()},
+    ],
+)
+@pytest.mark.parametrize("datasets", [{}, {OBJECTIVE: empty_dataset([1], [1])}])
+def test_qhsri_raises_for_invalid_models_keys(
+    datasets: dict[str, Dataset], models: dict[str, ProbabilisticModel]
+) -> None:
+    search_space = Box([-1], [1])
+    rule = BatchHypervolumeSharpeRatioIndicator()
+    with pytest.raises(ValueError):
+        rule.acquire(search_space, models, datasets=datasets)
+
+
+@pytest.mark.parametrize("models", [{}, {OBJECTIVE: QuadraticMeanAndRBFKernel()}])
+@pytest.mark.parametrize(
+    "datasets",
+    [
+        {},
+        {"foo": empty_dataset([1], [1])},
+        {"foo": empty_dataset([1], [1]), OBJECTIVE: empty_dataset([1], [1])},
+    ],
+)
+def test_qhsri_raises_for_invalid_dataset_keys(
+    datasets: dict[str, Dataset], models: dict[str, ProbabilisticModel]
+) -> None:
+    search_space = Box([-1], [1])
+    rule = BatchHypervolumeSharpeRatioIndicator()
+    with pytest.raises(ValueError):
+        rule.acquire(search_space, models, datasets=datasets)
+
+
+# @pytest.mark.parametrize("num_query_points", [5, 10])
```

Comment:
WIP or are you stuck here?

Also don't forget to add an integration test (which is the proof of the pudding). Feel free to ask me if you have any questions.

Reply:
Ah yeah, I couldn't get it to work because I was getting: ` NotImplementedError: Model QuadraticMeanAndRBFKernel() does not support predicting observations, just the latent function`, and I wasn't sure of the best way to proceed, whether to use a different kernel, or add a dummy prediction to that kernel.